src/Strategy/Python/Pipenv.hs

```diff
@@ -180,6 +193,9 @@ buildGraph lock maybeDeps = run . withLabeling toDependency $ do
             , dependencyTags = Map.empty
             }
 
+    -- We only have edges if we have graph dependencies. Only prune unreachable in this case
+    prune = if isJust maybeDeps then pruneUnreachable else id
```

Comment:
[nit] I'm not sure this guard is really worth it. If there are no graph dependencies `pruneUnreachable` can't really do anything can it?

Reply:
This is necessary: if there are no edges, then `pruneUnreachable` will remove all dependencies that aren't direct dependencies as there's no path to them.

Reply:
It's been a while since I've been deep in this code so forgive me - but in what cases would we actually prune anything then? If a dep is not direct, then it is by definition deep, right? 

Or is the distinction here that we need to handle both graphs that have any edges at all (and these should be subject to pruning) but sometimes we'll get a graph with no edges at all in which case we don't want to attempt pruning?

Reply:
> we need to handle both graphs that have any edges at all (and these should be subject to pruning) but sometimes we'll get a graph with no edges at all in which case we don't want to attempt pruning?

Exactly this. In the case of Pipenv, running with `--static-only-analysis` results in a graph with no edges and so we don't want to attemput pruning in this case.

Reply:
~This function is called `getDepsStatically`, is that name appropriate when it sounds like it can get graphs from a non-static source?~

Would you mind adding some commentary around how the cases where we want to prune vs. the ones where don't arise? Because this code is sort of operating at the level of the graph, I had trouble connecting it in my head to the business case it was handling.

Reply:
The function where this is happening is called`buildGraph`. I will expand on the comment